Automattic/pocket-casts-ios
# [Listening History] Display Search bar

| 📘 Part of: #2181 
|:---:|

Fixes #2191 
[Android ref](https://github.com/Automattic/pocket-casts-android/pull/2793)

This adds a search bar on the listening history screen.

## To test

> [!NOTE]
> make sure to enable the feature flag `listeningHistorySearch`

1. Login with an account with sufficient listening history
2. Go to Profile tab -> Listening History
3. ✅ Notice that a search bar now exists at the top
4. Long press an episode
5. ✅ Notice that multi-select bar is displayed at the top
6. Dismiss the multi-select toolbar
7. Scroll the listening history list
8. ✅ Notice that search bar scrolls behind the toolbar
9. Go to Profile tab -> Starred
10. ✅ Notice that search bar is not present on the screen
11. Go to Profile tab -> Downloads
12. ✅ Notice that search bar is not present on the screen
13. Disable feature flag
14. Go to Profile tab -> Listening History
15. ✅ Notice that search bar is not present on the screen

## Checklist
